### ISSUE TITLE
Duplicate fields in JSON

### ISSUE THREAD

**Issue by lexaknyazev** [KhronosGroup/glTF#846]
As pointed out by @andreasplesch in #845:
> Also, I did not find a statement regarding duplicate keys in the json: last one wins ?

**tl;dr:** 1.0 spec is silent on this. For glTF 2.0, duplicate keys aren't allowed.

---

According to ECMA-262 definition of `JSON.parse` ([5.1](https://www.ecma-international.org/ecma-262/5.1/#sec-15.12.2), [6.0](http://www.ecma-international.org/ecma-262/6.0/#sec-internalizejsonproperty), [7.0](http://www.ecma-international.org/ecma-262/7.0/index.html#sec-internalizejsonproperty)):
> In the case where there are duplicate name Strings within an object, lexically preceding values for the same key shall be overwritten.

But [RFC 7159, Section 4](https://tools.ietf.org/html/rfc7159#section-4) gives a bit different perspective:

> The names within an object SHOULD be unique.
>...
> An object whose names are all unique is interoperable in the sense that all software implementations receiving that object will agree on the name-value mappings. When the names within an object are not unique, the behavior of software that receives such an object is unpredictable. Many implementations report the last name/value pair only. Other implementations report an error or fail to parse the object, and some implementations report all of the name/value pairs, including duplicates.

Because glTF 2.0 usage goes beyond Web, it would be reasonable to require uniqueness of names.

CC @pjcozzi

**Comment by andreasplesch** [comment]
Thanks. As a side note, since it is not really possible to disallow people to attempt to use duplicate keys, it may be more accurate to say that this leads to undefined behaviour.

**Comment by pjcozzi** [comment]
> Because glTF 2.0 usage goes beyond Web, it would be reasonable to require uniqueness of names.

OK with me.

> As a side note, since it is not really possible to disallow people to attempt to use duplicate keys, it may be more accurate to say that this leads to undefined behaviour.

The validator could detect this.

**Comment by pjcozzi** [comment]
Updated in #826
